src/System.Runtime.Extensions/src/System/Runtime/CompilerServices/SwitchExpressionException.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Serialization;
+
+namespace System.Runtime.CompilerServices
+{
+    /// <summary>
+    /// Indicates that a switch expression that was non-exhaustive failed to match its input
+    /// at runtime, e.g. in the C# 8 expression <code>3 switch { 4 => 5 }</code>.
+    /// The exception optionally contains an object representing the unmatched value.
+    /// </summary>
+    [Serializable]
+    [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
+    public sealed class SwitchExpressionException : InvalidOperationException//, ISerializable
```

Comment:
Why the "//, ISerializable"?

Reply:
`ISerializable` is inherited -- as a matter of style, do we normally repeat interface derivations on derived classes?

Reply:
> do we normally repeat interface derivations on derived classes?

No